cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -152,6 +161,19 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
     if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
       authorizationEnforcer.enforce(instanceId, principal, Action.ADMIN);
       authorizer.grant(namespace, principal, ImmutableSet.of(Action.ALL));
+      // Also grant the namespace user all privileges on the namespace, if kerberos is enabled
+      if (kerberosEnabled) {
+        ImpersonationInfo impersonationInfo = impersonationUserResolver.getImpersonationInfo(metadata);
+        String namespacePrincipal = impersonationInfo.getPrincipal();
+        Preconditions.checkNotNull(
```

Comment:
I think the `getImpersonationInfo` method already guaranteed it won't be `null`, isn't it? Why checks it again?